### PR TITLE
[docs] cleanup README and add coverage badge

### PR DESCRIPTION
**Pull request by geekflyer**
### Description

This cleans up the readme of unused stuff and adds a working coverage badge

### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2936)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,11 +16,10 @@ Aptos-core strives towards being the safest and most scalable layer one blockcha
 * [Aptos Labs](https://aptoslabs.com/)
 * [Aptos Developer Network](https://aptos.dev)
 * [Getting Started](https://aptos.dev/guides/getting-started)
-* [Life of a Transaction](https://aptos.dev/guides/basics-life-of-txn)
```
Comment (davidiw) [inline README.md:22]: Hmm?

**Review by davidiw — COMMENTED** [review]
Fine but I don't know why we'd get rid of the pointer there...

nor get rid of the rust create docs?

**Comment by geekflyer** [comment]
> Fine but I don't know why we'd get rid of the pointer there...
> 
> nor get rid of the rust create docs?

the rust crate docs haven't been updated in months (still contain diem names etc.).

File: README.md (hunk 1 above)
Comment (geekflyer) [reply README.md:22]: I think the Life of a Transaction link on the README is kinda random. We already have a link to the docs entrypoint and to Getting Started. What makes the "Life of a Transaction" page so special that it needs to be mentioned here?

**Comment by davidiw** [comment]
> > Fine but I don't know why we'd get rid of the pointer there...
> > nor get rid of the rust create docs?
> 
> the rust crate docs haven't been updated in months (still contain diem names etc.).

Could you update them rather than deleting it?

File: README.md (hunk 1 above)
Comment (davidiw) [reply README.md:22]: Yeah, I'm not as opinionated about this, but I do think we shouldn't drop our docs pointer, we should take it as a point of action to 1) update it and 2) automate it.

**Comment by geekflyer** [comment]
> > > Fine but I don't know why we'd get rid of the pointer there...
> > > nor get rid of the rust create docs?
> > 
> > 
> > the rust crate docs haven't been updated in months (still contain diem names etc.).
> 
> Could you update them rather than deleting it?

Sure I could, but honestly I think is very low on my priority list (I think updating once isn't enough, we would have to automate this), so I would rather just delete the broken link and solve this properly later. I'm also not sure how much value crate docs provide given that aptos-core is not a library?

**Review by davidiw — COMMENTED** [review]
@banool / @gregnazario who may be working on resurrecting the Rust SDK.

**Comment by geekflyer** [comment]
> @banool / @gregnazario who may be working on resurrecting the Rust SDK.

I think once they actually get to resurrect the Rust SDK they can add some kind of docs auto-publishing back (which probably should publish to crates.io instead of github pages). For now I think it's better to simply delete the broken link so as to not confuse our users.

File: README.md [hunk 2]
```diff
@@ -4,12 +4,9 @@
 
 ---
 
-[![Aptos Rust Crate Documentation (main)](https://img.shields.io/badge/docs-main-59f)](https://aptos-labs.github.io/aptos-core/)
```
Comment (gregnazario) [inline README.md:7]: I think we should figure out how to build the docs rather than just remove them

**Review by davidiw — APPROVED** [review]

**Pull request merged**